crates/index-scheduler/src/scheduler/process_upgrade/mod.rs

```diff
@@ -0,0 +1,40 @@
+use meilisearch_types::milli;
+use meilisearch_types::milli::progress::{Progress, VariableNameStep};
+
+use crate::processing::UpgradeDatabaseProgress;
+use crate::{Error, IndexScheduler, Result};
+
+impl IndexScheduler {
+    pub(super) fn process_upgrade(&self, progress: Progress) -> Result<()> {
+        progress.update_progress(UpgradeDatabaseProgress::EnsuringCorrectnessOfTheSwap);
```

Comment:
this might require a change

Reply:
Actually I don't think we have to write anything here, I got rid of this type entirely